hooks/shapediver/viewer/interaction/gumball/useGumballEvents.ts

```diff
@@ -0,0 +1,89 @@
+import {
+	addListener,
+	EVENTTYPE_GUMBALL,
+	removeListener
+} from "@shapediver/viewer";
+import { GumballEventResponseMapping } from "@shapediver/viewer.features.gumball";
+import { useEffect, useState } from "react";
+
+// #region Functions (1)
+
+/**
+ * Hook allowing to create the gumball events.
+ * In this event handler, the transformed nodes are updated.
+ * 
+ * @param selectedNodes The selected nodes.
+ * @param initialTransformedNodeNames The initial transformed node names (used to initialize the selection state).
+ * 					Note that this initial state is not checked against the filter pattern. 
+ */
+export function useGumballEvents(
+	selectedNodeNames: string[],
+	initialTransformedNodeNames?: { name: string, transformation: number[] }[]
+): {
+    /**
+     * The transformed nodes.
+     */
+    transformedNodeNames: { name: string, transformation: number[] }[],
+    /**
+     * Set the transformed nodes.
+     * 
+     * @param nodes 
+     * @returns 
+     */
+    setTransformedNodeNames: (nodes: { name: string, transformation: number[] }[]) => void
+} {
+
+	// state for the transformed nodes
+	const [transformedNodeNames, setTransformedNodeNames] = useState<{ name: string, transformation: number[] }[]>(initialTransformedNodeNames ?? []);
+
+	// register an event handler and listen for output updates
+	useEffect(() => {
+		const token = addListener(EVENTTYPE_GUMBALL.MATRIX_CHANGED, (e) => {
+			const gumballEvent = e as GumballEventResponseMapping[EVENTTYPE_GUMBALL.MATRIX_CHANGED];
+
+			// Create a new array to avoid mutating the state directly
+			const newTransformedNodeNames = [...transformedNodeNames];
```

Comment:
@MajorMeerkatThe3rd the use of `transformedNodeNames` (which is not a dependency, and ideally wouldn't be made a dependency), could lead to trouble, depending on how this hook gets used. Example: If `setTransformedNodeNames` gets called by a caller of the `useGumballEvents` hook, then the gumball event callback will still use the old instance of `transformedNodeNames`. 
Potential solution: Also define a `useRef` below the definition of `transformedNodeNames, set TransformedNodeNames`, and always update its value to `transformedNodeNames`. Use the reference in the event handler callback.